src/seedu/addressbook/Main.java

```diff
@@ -28,10 +28,14 @@
     private TextUi ui;
     private StorageFile storage;
     private AddressBook addressBook;
+    private Parser parser; // Store as variable rather than always instantiating it
 
     /** The list of person shown to the user most recently.  */
     private List<? extends ReadOnlyPerson> lastShownList = Collections.emptyList();
 
+    public Main() {
+        this.parser = new Parser();
```

Comment:
Take note: changes that does not belongs to the PR / part of the features should be omitted.